libraries/chain/util/smt_token.cpp

```diff
@@ -68,6 +69,253 @@ fc::optional< time_point_sec > last_emission_time( const database& db, const ass
    return {};
 }
 
+namespace generation_unit {
+
+bool is_founder( const account_name_type& unit_target )
+{
+   return unit_target != SMT_DESTINATION_FROM && unit_target != SMT_DESTINATION_FROM_VESTING;
+}
+
+bool is_contributor( const account_name_type& unit_target )
+{
+   return !is_founder( unit_target );
+}
+
+bool is_vesting( const account_name_type& name )
+{
+   if ( name == SMT_DESTINATION_FROM_VESTING )
+      return true;
+   if ( name == SMT_DESTINATION_VESTING )
+      return true;
+
+   return false;
+}
+
+} // steem::chain::util::smt::generation_unit
+
+namespace ico {
+
+void payout( database& db, const asset_symbol_type& symbol, const account_object& account, const std::vector< asset >& assets )
+{
+   for ( auto& _asset : assets )
+   {
+      if ( _asset.symbol.is_vesting() )
+      {
+         db.create_vesting( account, asset( _asset.amount, _asset.symbol.get_paired_symbol() ) );
+      }
+      else
+      {
+         db.adjust_balance( account, _asset );
+
+         if ( _asset.symbol.space() == asset_symbol_type::smt_nai_space )
+            db.adjust_supply( _asset );
+      }
+
+      if ( _asset.symbol == STEEM_SYMBOL )
+      {
+         db.modify( db.get< smt_ico_object, by_symbol >( symbol ), [&]( smt_ico_object& obj )
+         {
+            obj.processed_contributions += _asset.amount;
+         } );
+      }
+   }
+}
+
+bool schedule_next_refund( database& db, const asset_symbol_type& a )
+{
+   bool action_scheduled = false;
+   auto& idx = db.get_index< smt_contribution_index, by_symbol_id >();
+   auto itr = idx.lower_bound( boost::make_tuple( a, 0 ) );
```

Comment:
This doesn't need to change, but in the future, you shouldn't need to create a tuple with an id of 0. Passing in just the asset symbol type should do the same thing.